src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6549,14 +6554,29 @@ public override IOperation VisitTranslatedQuery(ITranslatedQueryOperation operat
 
         public override IOperation VisitConstantPattern(IConstantPatternOperation operation, int? captureIdForResult)
         {
-            return new ConstantPatternOperation(Visit(operation.Value), semanticModel: null,
+            return new ConstantPatternOperation(operation.InputType, Visit(operation.Value), semanticModel: null,
                 syntax: operation.Syntax, isImplicit: IsImplicit(operation));
         }
 
         public override IOperation VisitDeclarationPattern(IDeclarationPatternOperation operation, int? captureIdForResult)
         {
-            return new DeclarationPatternOperation(operation.DeclaredSymbol, semanticModel: null,
-                operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+            return new DeclarationPatternOperation(
+                inputType: operation.InputType, matchedType: operation.MatchedType, operation.DeclaredSymbol, operation.MatchesNull, semanticModel: null,
+                operation.Syntax, IsImplicit(operation));
+        }
+
+        public override IOperation VisitRecursivePattern(IRecursivePatternOperation operation, int? argument)
+        {
+            return new RecursivePatternOperation(
+                inputType: operation.InputType,
+                matchedType: operation.MatchedType,
+                operation.DeconstructSymbol,
+                operation.DeconstructionSubpatterns.SelectAsArray(p => (IPatternOperation)Visit(p)),
```

Comment:
Can these never have control flow, even in error scenarios? ie, what would the ioperation tree look like if I were to write something like `e switch { Foo(a: b ?? c) => 1 }` (may not have gotten the syntax right, but the general idea is some non-constant pattern that the compiler will parse and bind, even if it errored)?

Reply:
In this PR, the switch expression is not supported in the control flow graph.

---
In reply to: [244199574](https://github.com/dotnet/roslyn/pull/32018#discussion_r244199574) [](ancestors = 244199574)